src/Raven.Server/Documents/Document.cs

```diff
@@ -102,37 +104,55 @@ public bool Expired(DateTime currentDate)
             return false;
         }
 
-        private static bool IsMetadataEqualTo(BlittableJsonReaderObject currentDocument, BlittableJsonReaderObject targetDocument)
+        private static DocumentCompareResult IsMetadataEqualTo(BlittableJsonReaderObject currentDocument, BlittableJsonReaderObject targetDocument, bool skipAttachment)
```

Comment:
Yes, it looks like we can merge those two blocks of code. 
I think that DocumentComparison woulc be a better name here, as we check for metadata first which means that we work with documents.